engine/src/python/base_computer/ship_view.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * ship_view.h
+ *
+ * Copyright (c) 2001-2002 Daniel Horn
+ * Copyright (c) 2002-2019 pyramid3d and other Vega Strike Contributors
+ * Copyright (c) 2019-2023 Stephen G. Tuggy, Benjamen R. Meyer, Roy Falk and other Vega Strike Contributors
+ *
+ * https://github.com/vegastrike/Vega-Strike-Engine-Source
+ *
+ * This file is part of Vega Strike.
+ *
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Vega Strike is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// -*- mode: c++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+#ifndef VEGA_STRIKE_ENGINE_PYTHON_BASE_COMPUTER_SHIP_VIEW_H
+#define VEGA_STRIKE_ENGINE_PYTHON_BASE_COMPUTER_SHIP_VIEW_H
+
+#include <string>
+#include <map>
+
+const std::string GetShipView(const std::map<std::string, std::string>ship_stats);
```

Comment:
this should probably be a const reference:
```
const std::string GetShipView(const std::map<std::string, std::string>& ship_stats);
```
otherwise it's going to copy the map.

Reply:
Answering both your questions here:
1. This is for a single ship, not all ships. It will be used by ship view - that page in base computer where you see the ship specs. It will also be used by save ship as part of the save game. If you have a fleet of ships, it will several times.
2. It will copy. I'll fix it as it is good practice. However, it is negligible.

Reply:
Thanks!